Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -401,30 +481,97 @@ void createSeedsForGroupSycl(
       // *********** TRIPLET SEARCH - BEGIN *********** //
       //************************************************//
 
-      const auto maxMemoryAllocation = std::min(
-          edgesComb, globalBufferSize / uint32_t((sizeof(detail::TripletData) +
-                                                  sizeof(detail::SeedData)) *
-                                                 2));
+      // EXPLANATION OF INDEXING (second part)
+      /*
+        For the triplet search, we calculate the upper limit of constructible
+        triplets.
+
+        For this, we multiply the number of compatible bottom and compatible
+        top SPs for each middle SP, and add these together.
+        (nb0*nt0 + nb1*nt1 + ... where nbk is the number of compatible bottom
+        SPs for the kth middle SP, similarly ntb is for tops)
+
+        sumBotTopCombined is a prefix sum array (of length M+1) of the
+        calculated combinations.
 
-      detail::TripletData* deviceCurvImpact =
-          cl::sycl::malloc_device<detail::TripletData>(maxMemoryAllocation,
-                                                       (*q));
+        sumBotTopCombined:
+        ________________________________________________________
+        |     |         |                   |     |  M         | M = number
+        |  0  | nb0*nt0 | nb0*nt0 + nb1*nt1 | ... |  ∑ nbi+nti | of middle
+        |_____|_________|___________________|_____|_i=0________| space points
+
+        We will start kernels and reserve memory for these combinations but
+        only so much we can fit into memory at once.
+
+        We limit our memory usage to globalBufferSize/2, this is currently
```

Comment:
why is this /2 and not the full buffer size?

Reply:
This is not an optimal solution. What we have here, is not the unallocated global memory size, but the whole. We need space for other allocations as well, so we can't use the whole.

We could do two things:
1. ask for still available memory size (I didn't find a way to do that in SYCL)
2. keep track of allocated memory size, and extract that from the whole (I'm not sure how padding is added in this case)

It just seemed to be an easier way, to use /2 of buffer size.

For later, it would be better to do something similar to what Attila did, I left a note about that.
I wouldn't stress too much about this, because it is not optimal either way.

Reply:
Third solution, what Attila does (simplified):

Allocate memory in once (and free it in once too), and manually increment pointers.